client/src/app/components/nova/address/section/native-tokens/Asset.tsx

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable jsdoc/require-param */
+/* eslint-disable jsdoc/require-returns */
+import { FoundryOutput, MetadataFeature, FeatureType, Irc30Metadata } from "@iota/sdk-wasm-nova/web";
+import { Validator as JsonSchemaValidator } from "jsonschema";
+import React, { ReactElement, useEffect, useState } from "react";
+import { AssetProps } from "./AssetProps";
+import tokenSchemeIRC30 from "~assets/schemas/token-schema-IRC30.json";
+import { useFoundryDetails } from "~helpers/stardust/hooks/useFoundryDetails";
+import { useTokenRegistryNativeTokenCheck } from "~helpers/stardust/hooks/useTokenRegistryNativeTokenCheck";
+import { Converter } from "~helpers/stardust/convertUtils";
+import Spinner from "../../../../Spinner";
+import TruncatedId from "~/app/components/stardust/TruncatedId";
+import { useNetworkInfoNova } from "~/helpers/nova/networkInfo";
+
+/**
+ * Component which will display an asset.
+ */
+const Asset: React.FC<AssetProps> = ({ tableFormat, token }) => {
+    const { name: network } = useNetworkInfoNova((s) => s.networkInfo);
+    const [foundryDetails, isLoading] = useFoundryDetails(network, token.id);
+    const [tokenMetadata, setTokenMetadata] = useState<Irc30Metadata | null>(null);
+    const [isWhitelisted] = useTokenRegistryNativeTokenCheck(token.id);
+
+    useEffect(() => {
+        if (isWhitelisted && foundryDetails) {
+            const immutableFeatures = (foundryDetails?.output as unknown as FoundryOutput).immutableFeatures;
+
+            const metadata = immutableFeatures?.find((feature) => feature.type === FeatureType.Metadata) as MetadataFeature;
+
+            if (metadata) {
+                updateTokenInfo(metadata);
+            }
+        }
+    }, [isWhitelisted, foundryDetails]);
+
+    const updateTokenInfo = (metadata: MetadataFeature): void => {
+        const validator = new JsonSchemaValidator();
+
+        try {
+            const tokenInfo = JSON.parse(Converter.hexToUtf8(metadata.data)) as Irc30Metadata;
```

Comment:
Let's use the `hexToUtf8` from SDK nova here (to move away from the iota.js stuff).